packages/firebase_ml_vision/lib/src/face_detector.dart

```diff
@@ -4,18 +4,229 @@
+        'enableTracking': options.enableTracking,
+        'minFaceSize': options.minFaceSize,
+        'mode': _enumToString(options.mode),
+      },
+    );
+
+    final List<Face> faces = <Face>[];
+    for (dynamic data in reply) {
+      faces.add(Face._(data));
+    }
+
+    return faces;
   }
 }
 
-class FaceDetectorOptions {}
+/// Options for [FaceDetector].
+///
+/// Used to configure the features of the [FaceDetector], such as
+/// classification, face tracking, speed, etc.
+class FaceDetectorOptions {
+  /// Constructor for [FaceDetectorOptions].
+  ///
+  /// The parameter minFaceValue must be between 0.0 and 1.0, inclusive.
+  FaceDetectorOptions({
+    this.enableClassification = false,
+    this.enableLandmarks = false,
+    this.enableTracking = false,
+    this.minFaceSize = 0.1,
+    this.mode = FaceDetectorMode.fast,
+  })  : assert(minFaceSize >= 0.0),
+        assert(minFaceSize <= 1.0);
+
+  /// Whether to run additional classifiers for characterizing attributes.
+  ///
+  /// E.g. "smiling" and "eyes open".
+  final bool enableClassification;
+
+  /// Whether to detect [FaceLandmark]s.
+  final bool enableLandmarks;
+
+  /// Whether to enable face tracking.
+  ///
+  /// If enabled, the detector will maintain a consistent ID for each face when
+  /// processing consecutive frames.
+  final bool enableTracking;
+
+  /// The smallest desired face size.
+  ///
+  /// Expressed as a proportion of the width of the head to the image width.
+  ///
+  /// Must be a value between 0.0 and 1.0.
+  final double minFaceSize;
+
+  /// Option for controlling additional accuracy / speed trade-offs.
+  final FaceDetectorMode mode;
+}
+
+/// Represents a face detected by [FaceDetector].
+class Face {
+  Face._(dynamic data)
+      : boundingBox = Rectangle<int>(
+          data['left'],
+          data['top'],
+          data['width'],
+          data['height'],
+        ),
+        headEulerAngleY = data['headEulerAngleY'],
+        headEulerAngleZ = data['headEulerAngleZ'],
+        leftEyeOpenProbability = data['leftEyeOpenProbability'],
+        rightEyeOpenProbability = data['rightEyeOpenProbability'],
+        smilingProbability = data['smilingProbability'],
+        trackingId = data['trackingId'],
+        _landmarks = Map<FaceLandmarkType, FaceLandmark>.fromIterables(
+          FaceLandmarkType.values,
+          List<FaceLandmark>.generate(
+            FaceLandmarkType.values.length,
+            (int index) {
+              final FaceLandmarkType type = FaceLandmarkType.values[index];
+              final dynamic landmarkMap = data['landmarks'];
+              final String typeString = _enumToString(type);
+
+              if (landmarkMap[typeString] == null) return null;
+              return FaceLandmark._(
+                type,
+                Point<int>(
+                  landmarkMap[typeString][0],
+                  landmarkMap[typeString][1],
+                ),
+              );
+            },
+          ),
```

Comment:
If you have a `list` of type `List<T>` and an expression `expr(t)` that has type `U` when `t` has type `T`, then you can replace
```dart
List<U>.generate(list.length, (int index) => expr(list[index]))
```
by the simpler
```dart
list.map<U>((T t) => expr(t)).toList()
```
provided `expr` does not use `index`. Since you only need an `Iterable<U>`, you can dispense with the `.toList()` and simplify your code to
```dart
_landmarks = Map<FaceLandmarkType, FaceLandmark>.fromIterables(
  FaceLandmarkType.values,
  FaceLandmarkType.values.map<FaceLandmark>((FaceLandmarkType type) {
    final dynamic landmarkMap = data['landmarks'];
    final String typeString = _enumToString(type);
    if (landmarkMap[typeString] == null) return null;
    return FaceLandmark._(
      type,
      Point<int>(
        landmarkMap[typeString][0],
        landmarkMap[typeString][1],
      ),
    );
  }
);
```
Next, you can avoid repeated lookup of `data['landmarks']` and `landmarkMap[typeString]` by a bit of hoisting:
```dart
final Map<dynamic, dynamic> landmarkMap = data['landmarks'];
_landmarks = Map<FaceLandmarkType, FaceLandmark>.fromIterables(
  FaceLandmarkType.values,
  FaceLandmarkType.values.map<FaceLandmark>((FaceLandmarkType type) {
    final String typeString = _enumToString(type);
    final List<dynamic> position = landmarkMap[typeString];
    return (position == null) ? null : FaceLandmark._(type, Point<int>(position[0], position[1]));
  }
);
```